File: android/app/src/main/java/dev/pepotech/pepomote/control/UiSounds.kt
package dev.pepotech.pepomote.control

import android.content.Context
import android.media.AudioAttributes
import android.media.SoundPool
import dev.pepotech.pepomote.R

/** Sonidos UI sintetizados propios (assets/sounds-src/gen_sounds.py). */
object UiSounds {
    private var pool: SoundPool? = null
    private var blipId = 0
    private var popId = 0
    private var tickId = 0
    private var connectId = 0
    private var disconnectId = 0

    /** Interruptor global (Ajustes). La háptica no se toca, solo el audio. */
    @Volatile
    var enabled = true

    fun init(context: Context) {
        if (pool != null) return
        enabled = AppPrefs.soundsEnabled(context)
        val attrs = AudioAttributes.Builder()
            .setUsage(AudioAttributes.USAGE_ASSISTANCE_SONIFICATION)
            .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
            .build()
        val p = SoundPool.Builder().setMaxStreams(4).setAudioAttributes(attrs).build()
        val app = context.applicationContext
        blipId = p.load(app, R.raw.ui_blip, 1)
        popId = p.load(app, R.raw.ui_pop, 1)
        tickId = p.load(app, R.raw.ui_tick, 1)
        connectId = p.load(app, R.raw.ui_connect, 1)
        disconnectId = p.load(app, R.raw.ui_disconnect, 1)
        pool = p
    }

    private fun play(id: Int, vol: Float = 0.8f) {
        if (!enabled) return
        pool?.play(id, vol, vol, 1, 0, 1f)
    }

    fun blip() = play(blipId)
    fun pop() = play(popId)
    fun tick() = play(tickId)
    fun connect() = play(connectId)
    fun disconnect() = play(disconnectId, 0.6f)
}

File: ios/PepoMote/Control/UiSounds.swift
import AVFoundation
import Foundation

/// Sonidos UI sintetizados propios (los mismos WAV que Android,
/// assets/sounds-src/gen_sounds.py). Se mezclan con la música del sistema y
/// no la interrumpen.
final class UiSounds {
    static let shared = UiSounds()

    /// Interruptor global (Ajustes). La háptica no se toca, solo el audio.
    var enabled = true

    private var players: [String: AVAudioPlayer] = [:]
    private var ready = false

    func prepare() {
        if ready { return }
        ready = true
        enabled = AppPrefs.soundsEnabled
        do {
            try AVAudioSession.sharedInstance().setCategory(.ambient, options: [.mixWithOthers])
            try AVAudioSession.sharedInstance().setActive(true)
        } catch {
            // sin sesión de audio: la app sigue, sin sonidos
        }
        for name in ["ui_blip", "ui_pop", "ui_tick", "ui_connect", "ui_disconnect"] {
            guard let url = Bundle.main.url(forResource: name, withExtension: "wav") else { continue }
            if let p = try? AVAudioPlayer(contentsOf: url) {
                p.prepareToPlay()
                players[name] = p
            }
        }
    }

    private func play(_ name: String, _ vol: Float = 0.8) {
        guard enabled, let p = players[name] else { return }
        p.volume = vol
        p.currentTime = 0
        p.play()
    }

    func blip() { play("ui_blip") }
    func pop() { play("ui_pop") }
    func tick() { play("ui_tick") }
    func connect() { play("ui_connect") }
    func disconnect() { play("ui_disconnect", 0.6) }
}
